api-spec/openapi/STAC.yaml

```diff
@@ -328,6 +319,21 @@ components:
           example: 'http://www.geoserver.example/stac/naip/child/catalog.json'
         rel:
           example: child
+        method:
+          type: string
+          enum:
+            - GET
+            - POST
+          default: get
+          description: Specifies the HTTP method that the link expects
+        headers:
+          type: object
+          description: Object key values pairs they map to headers
+          example:
+            Accept:  application/json
+        body:
+          type: string
+          description: For POST requests, the link can specify the HTTP body as a JSON object.
```

Comment:
How well does this handle www-formencoded and form/multipart?

Reply:
I think formencoded should be fine, but multipart might be a little difficult. 

I don't think multipart has an obvious JSON representation, but perhaps we can lean on existing tools? It seems many client side libraries make use of https://github.com/form-data/form-data

Of particular interest is the request library that provides supports formData as an object. 

I don't think it would solve the file streaming issue, but it would allow for basic multipart support. Also, I would recommend that any file transfers should be referenced for additional page requests rather than resubmitted. It would be in the servers control to make that decision with this approach.

We might want to consider making body an object instead with some rules for how different content types are supported? application/json  would just be passed through for example.

Reply:
I thought we decided that the body would be "string"

Reply:
Sorry, that was a change when discussing with Phil. He prefers that the key value pairs be expressed as JSON and the client would know how to apply them to the body. I'm happy to facilitate more discussion, but I'm not sure I feel strongly either way. Let's get a discussion going today.